src/app/(main)/my-voyage/[teamId]/ideation/components/IdeationForm.tsx

```diff
@@ -168,39 +171,110 @@ export default function IdeationForm() {
     () => () => {
       void persistor.purge();
     },
-    [],
+    []
   );
 
-  function renderButtonContent() {
-    if (editIdeationLoading || addIdeationLoading) {
-      return <Spinner />;
-    }
+  // This block is responsible for auto-save functionality. Right now nextjs does
+  // not have a way to intercept routes with app router. When that is implemented
+  // on their side, it will probably be better to go that method.
 
-    return editMode ? "Save Changes" : "Save";
+  function asyncTimeout(ms: number) {
+    return new Promise((resolve) => {
+      setSaveTimeout(setTimeout(resolve, ms));
+    });
   }
 
-  function renderDeleteButtonContent() {
-    if (deleteIdeationLoading) {
+  useEffect(() => {
+    async function autoSave() {
```

Comment:
I guess we're gonna have this `autoSave` functionality in many places. Is it possible to move it to utils so we can just import and call it from other places (passing some arguments of course)?

Reply:
So basically everytime a user types smth, it gets saved to the database (every 5 seconds)... isn't it too many API calls?

Reply:
Yes ideally we want this as a re-usable function. The reason I didn't do that in this PR is because there's quite a lot going on in this logic and Idk what will be needed / not needed for the other features so I couldn't really come up with a good refactor for this right now. I think we can revisit this in the future when we have a clearer idea of what's common between the different forms.

As for saving to the backend. Yes, that is an issue. However, we don't have a better solution right now unfortunately. As soon as client-side route interceptors get supported, we should migrate to that solution instead of this.